### PR TITLE
Fix various smaller bugs

### PR DESCRIPTION
**Pull request by mfherbst**
Fixes for smaller bugs I triggered while playing around today.

### CODE REVIEW

File: src/eigen/lobpcg_hyper_impl.jl [hunk 1]
```diff
@@ -118,7 +118,7 @@ end
 # Perform a Rayleigh-Ritz for the N first eigenvectors.
 @timing function rayleigh_ritz(X, AX, N)
     XAX = X' * AX
-    @assert all(!isnan, XAX)
+    @assert !any(isnan, XAX)
```
Comment (antoine-levitt) [inline src/eigen/lobpcg_hyper_impl.jl:121]: any particular reason for this?

Comment (mfherbst) [reply src/eigen/lobpcg_hyper_impl.jl:121]: Yes. On cuda funnily the two statements are not equal. I had the original assertion fail even without nans in the array.

Comment (antoine-levitt) [reply src/eigen/lobpcg_hyper_impl.jl:121]: What? Sounds like a bug that should be reported?

Comment (mfherbst) [reply src/eigen/lobpcg_hyper_impl.jl:121]: Yeah, probably. In any case we very easily get NaNs when using CUDA, so we need to give this a closer look anyway.

**Pull request merged**